packages/resin-mixpanel-client/src/resin-mixpanel-client.js

```diff
@@ -6,113 +7,124 @@ module.exports = function(token) {
+				// `identify` is called before continuing to tracking
+				return self.login(uid)
+			})
 		},
-		login: function(uid, callback) {
-			if (isBrowser) {
-				mixpanel().identify(uid)
-			}
+		login: function(uid) {
 			userId = uid
-			if (typeof callback === "function") callback()
-		},
-		logout: function(callback) {
-			var ref
+
 			if (isBrowser) {
-				if ((ref = mixpanel().cookie) != null) {
-					ref.clear()
-				}
+				return mixpanel.then(function(mp) {
+					mp.identify(uid)
+				})
 			}
-			userId = null
-			if (typeof callback === "function") callback()
+
+			return Promise.resolve()
 		},
-		set: function(props, callback) {
+		logout: function() {
+			userId = null
+
 			if (isBrowser) {
-				mixpanel().register(props)
+				return mixpanel.then(function(mp) {
+					mp.reset()
+				})
 			}
 
-			if (typeof callback === "function") callback()
+			return Promise.resolve()
 		},
-		setOnce: function(props, callback) {
+		set: function(props) {
 			if (isBrowser) {
-				mixpanel().register_once(props)
+				return mixpanel.then(function(mp) {
+					mp.register(props)
+				})
 			}
 
-			if (typeof callback === "function") callback()
+			return Promise.resolve()
 		},
-		setUser: function(prop, to, callback) {
-			var mp = mixpanel()
-
+		setOnce: function(props) {
 			if (isBrowser) {
-				if (!callback && typeof to === 'function') {
-					callback = wrapCallback(to)
-					return mp.people.set(prop, callback)
-				} else {
-					callback = wrapCallback(callback)
-					return mp.people.set(prop, to, callback)
-				}
-			} else {
-				if (!userId) {
-					throw new Error('(Resin Mixpanel Client) Please login() before using setUser()')
-				}
-				return mp.people.set(userId, prop, to, callback)
+				return mixpanel.then(function(mp) {
+					mp.register_once(props)
+				})
 			}
+
+			return Promise.resolve()
 		},
-		setUserOnce: function(prop, to, callback) {
-			var mp = mixpanel()
+		setUser: function(props) {
+			return mixpanel.then(function(mp) {
+				return Promise.fromCallback(function(callback) {
+					if (isBrowser) {
+						callback = wrapBrowserCallback(callback)
+						return mp.people.set(props, callback)
+					}
 
-			if (isBrowser) {
-				if (!callback && typeof to === 'function') {
-					callback = wrapCallback(to)
-					return mp.people.set_once(prop, callback)
-				} else {
-					callback = wrapCallback(callback)
-					return mp.people.set_once(prop, to, callback)
-				}
-			} else {
-				if (!userId) {
-					throw new Error('(Resin Mixpanel Client) Please login() before using setUserOnce()')
-				}
-				return mp.people.set_once(userId, prop, to, callback)
-			}
+					if (!userId) {
+						throw new Error('(Resin Mixpanel Client) Please login() before using setUser()')
+					}
+					return mp.people.set(userId, props, callback)
+				})
+			})
 		},
-		track: function(event, properties, callback) {
-			var mp = mixpanel()
+		setUserOnce: function(props, callback) {
```

Comment:
You can remove the `callback` parameter as it's unused and looks like it's always undefined.